src/scripts/sync_synonym.py

```diff
@@ -0,0 +1,177 @@
+            #  not filter by things we want out, but by the list of synonym preds we want in?
+            if pred == 'rdfs:label':
+                continue
+            tups.add((_id, pred, alias))
+    df = pd.DataFrame(tups, columns=['curie', 'synonym_scope', 'synonym'])
+    return df
+
+
+def _filter_non_intersecting_rows(base_df: pd.DataFrame, intersecting_df: pd.DataFrame, join_on: List[str]) -> pd.DataFrame:
+    """Filter out all rows are not in common, given a `join_on`. """
+    merge_df = base_df.merge(intersecting_df, on=join_on, how='left', indicator=True)
+    base_df2 = merge_df[merge_df['_merge'] == 'left_only'].drop(columns=['_merge']).rename(
+        columns=lambda x: x[:-2] if x.endswith('_x') else x)[list(base_df.columns)]
+    return base_df2
+
+
+# TODO
+def sync_synonyms(
+    ontology_db_path: Union[Path, str], mondo_scopes_and_xrefs_path: Union[Path, str],
+    mondo_mappings_path: Union[Path, str], onto_config_path: Union[Path, str], outpath_added: Union[Path, str],
+    outpath_confirmed: Union[Path, str], outpath_deleted: Union[Path, str], outpath_updated: Union[Path, str]
+):
+    """Create outputs for syncing synonyms between Mondo and its sources."""
+    # Iterate over mappings
+    mappings_df = pd.read_csv(mondo_mappings_path, sep='\t', comment='#')
+    # todo: refactor filtering by prefixes
+    #  check if this pattern has been used elsewhere; I think it has. and abstract into func for these 3 lines; 'filter prefixes'
+    #  - OR use this?: get_all_owned_terms()
+    #    - it seems long and messy compared to these 3 lines, but is more comprehensive. I could make get_all_owned_terms() a simple 3 line func and then make a separate func for all edge cases?
+    # - filter by source
+    owned_prefix_map: PREFIX_MAP = get_owned_prefix_map(onto_config_path)
+    mappings_df['object_prefix'] = mappings_df['object_id'].apply(lambda x: x.split(':')[0])
+    mappings_df = mappings_df[mappings_df['object_prefix'].isin(owned_prefix_map.keys())]
+    del mappings_df['object_prefix']
+
+    # Query synonyms
+    # - Mondo
+    mondo_df = pd.read_csv(mondo_scopes_and_xrefs_path, sep='\t').rename(columns={'?dbXref': 'source_id'})
+    mondo_df = mondo_df.rename(columns=lambda x: x.lstrip('?'))
+    mondo_df['object_prefix'] = mondo_df['source_id'].apply(lambda x: x.split(':')[0])
+    mondo_df = mondo_df[mondo_df['object_prefix'].isin(owned_prefix_map.keys())]
+    del mondo_df['object_prefix']
+    mondo_df['unprefixed_source_id'] = mondo_df['source_id'].apply(lambda x: x.split(':')[1])
+    mondo_df = mondo_df[mondo_df['unprefixed_source_id'].str.isnumeric()]
+    del mondo_df['unprefixed_source_id']
+
+    # - Source
+    source_db: SqlImplementation = get_adapter(ontology_db_path)
+    source_df: pd.DataFrame = _query_synonyms(mappings_df['object_id'].tolist(), source_db)
+    source_mondo_lookup = mappings_df.set_index('object_id')['subject_id'].to_dict()
```

Comment:
I'm assuming that Mondo terms will only ever be exact mapped to 1 and only 1 term from a given source. If you'd like me to do a QC check and raise an error if there's ever >1 exact mapping, let me know. Or mark resolved if unnecessary.

Reply:
@joeflack4 can you give an example of what you are assuming is the case ("Mondo terms will only ever be exact mapped to 1 and only 1 term from a given source") and also what does not fit that case?

Reply:
There are plenty of cases where the 1:1 assumption is wrong: "legal proxy merges". So the code needs to handle these.

Legal proxy merges are cases where 1 mondo id has more than 1 external exact match. In this case, the synonyms from both external terms travel to the 1 mondo id.

Reply:
@twhetzel I don't have a current example, nor anything I'm encountering which doesn't fit some case I'm expecting. I'm just checking on one of my assumptions. 

But my assumption was wrong! I forgot about legal proxy merges. I will update the code to handle >1 exact matches accordingly!